### PR TITLE
Various additions

### PR DESCRIPTION
**Pull request by SnivyLlama**
I added
- A way to save a profile on a specified interval, or when leaving the server
- A way to stop overeating in Auto Eat
- A mode in KillAura that is basically Any but with Auto Switch, will try to switch to sword or axe
- An option to rotate to the mined block in Nuker
- Allow dashes in the server ip list (some servers have dashes)
- Remove the name to player nametags
- Added CombatHUD labels to replace "Unknown"

### CODE REVIEW

File: src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/ProfilesTab.java [hunk 1]
```diff
@@ -147,6 +148,26 @@ public void initWidgets(Profile ogProfile, List<String> list) {
...
+            leaveSaveBool.action = () -> newProfile.saveOnLeave = leaveSaveBool.checked;
+            table.row();
+
+            // Interval
+            table.add(theme.label("Save on Interval"));
+            WCheckbox intervalSaveBool = table.add(theme.checkbox(ogProfile.saveOnInterval)).widget();
+            intervalSaveBool.action = () -> newProfile.saveOnInterval = intervalSaveBool.checked;
+            table.row();
+
+            table.add(theme.label("Interval Delay"));
```
Comment (SIMULATAN) [inline src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/ProfilesTab.java:163]: Inconsistency with the labels

File: src/main/java/meteordevelopment/meteorclient/systems/modules/player/AutoEat.java [hunk 2]
```diff
@@ -77,6 +77,13 @@
         .build()
     );
 
+    private final Setting<Boolean> avoidOvereating = sgHunger.add(new BoolSetting.Builder()
+        .name("avoid-overeating")
+        .description("Avoid eating foods that would bring your hunger over full.")
```
Comment (SIMULATAN) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/player/AutoEat.java:82]: I kinda dislike the naming, tbh I don't have a better replacement tho.
What about something like "that would make your saturation overshoot the hunger bar"?

**Review by SIMULATAN — CHANGES_REQUESTED** [review]

File: src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/ProfilesTab.java (hunk 1 above)
Comment (SnivyLlama) [reply src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/ProfilesTab.java:163]: Where is the inconsistency?

File: src/main/java/meteordevelopment/meteorclient/systems/modules/player/AutoEat.java (hunk 2 above)
Comment (SnivyLlama) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/player/AutoEat.java:82]: Not saturation though, you sure?

File: src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/ProfilesTab.java (hunk 1 above)
Comment (SnivyLlama) [reply src/main/java/meteordevelopment/meteorclient/gui/tabs/builtin/ProfilesTab.java:163]: Oh nvm its the colons

File: src/main/java/meteordevelopment/meteorclient/systems/modules/player/AutoEat.java (hunk 2 above)
Comment (SIMULATAN) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/player/AutoEat.java:82]: Maybe replace it with  "hunger level" then?

File: src/main/java/meteordevelopment/meteorclient/systems/hud/modules/CombatHud.java [hunk 3]
```diff
@@ -41,6 +38,7 @@
 public class CombatHud extends HudElement {
     private static final Color GREEN = new Color(15, 255, 15);
     private static final Color RED = new Color(255, 15, 15);
+    private static final Color YELLOW = new Color(255, 150, 15);
     private static final Color BLACK = new Color(0, 0, 0, 255);
```
Comment (SIMULATAN) [inline src/main/java/meteordevelopment/meteorclient/systems/hud/modules/CombatHud.java:42]: Constants are in the `Color` class now; https://github.com/MeteorDevelopment/meteor-client/blob/master/src/main/java/meteordevelopment/meteorclient/utils/render/color/Color.java#L16-L28
Pretty recent change so no wonder that it's not changed

Comment (SnivyLlama) [reply src/main/java/meteordevelopment/meteorclient/systems/hud/modules/CombatHud.java:42]: Hmm I'll check that out

**Comment by SnivyLlama** [comment]
There are so many useless commits that I'm not so sure this pr will be merged :pensive:

**Comment by SIMULATAN** [comment]
> There are so many useless commits that I'm not so sure this pr will be merged :pensive:

Again, squash merge exists

**Comment by SnivyLlama** [comment]
Oh

**Comment by RickyTheRacc** [comment]
@MineGame159 please 🙏

**Comment by SnivyLlama** [comment]
Fixed the conflict, might be worth merging just saying

File: src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java [hunk 4]
```diff
@@ -119,7 +129,17 @@ private void onTick(TickEvent.Pre event) {
...
+            if (!entities.get().getBoolean(e.getType())) continue;
+            if (ignoreOwn.get() && ((ProjectileEntityAccessor) e).getOwnerUuid().equals(mc.player.getUuid())) continue;
+
+            List<Box> futureArrowHitboxes = new ArrayList<>();
+
+            for (int i = 0; i < arrowLookahead.get(); i++) {
+                Vec3d nextPos = e.getPos().add(e.getVelocity().multiply(i / 5.0f));
+                futureArrowHitboxes.add(new Box(
+                    nextPos.subtract(e.getBoundingBox().getXLength() / 2, 0, e.getBoundingBox().getZLength() / 2),
+                    nextPos.add(e.getBoundingBox().getXLength() / 2, e.getBoundingBox().getYLength(), e.getBoundingBox().getZLength() / 2)));
```
Comment (C10udburst) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java:142]: this was not merged properly, lines 132 and 134-142 should be removed

File: src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java [hunk 5]
```diff
@@ -32,6 +34,14 @@
...
     private final SettingGroup sgMovement = settings.createGroup("Movement");
 
+    private final Setting<Integer> arrowLookahead = sgGeneral.add(new IntSetting.Builder()
+        .name("arrow-lookahead")
+        .description("How many steps into the future should be taken into consideration when deciding the direction")
+        .defaultValue(500)
+        .range(1, 750)
+        .build()
+    );
+
```
Comment (C10udburst) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java:44]: this also shouldn't be here

**Review by C10udburst — CHANGES_REQUESTED** [review]

**Comment by SnivyLlama** [comment]
tfw half the commits are upstream fetches
(squaaaash merge)

**Comment by arlomcwalter** [comment]
> tfw half the commits are upstream fetches
> 
> (squaaaash merge)

you know you don't have to make a commit to merge upstream

**Comment by SnivyLlama** [comment]
> > tfw half the commits are upstream fetches
> > (squaaaash merge)
> 
> you know you don't have to make a commit to merge upstream

yeah but its easier for me when compiling

**Comment by SnivyLlama** [comment]
oh wait i misread what you said how do you do that??

**Comment by arlomcwalter** [comment]
> oh wait i misread what you said how do you do that??

https://sdqweb.ipd.kit.edu/wiki/Git_pull_--rebase_vs._--merge

**Comment by arlomcwalter** [comment]
also as a very general rule, if you are contributing to a repo, make a branch per feature in your fork and submit one pull request per feature. this way you can delete feature branches after the pr is merged and maintain one clean master branch without a messed up commit log. (also read what i said above so you don't spam merge commits all the time.

**Pull request closed without merging**